tests/bsim/bluetooth/host/adv/extended/src/ext_adv_scanner.c

```diff
@@ -0,0 +1,195 @@
+/**
+ * Copyright (c) 2024 Croxel, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/kernel.h>
+
+#include "common.h"
+
+#include "bs_types.h"
+#include "bs_tracing.h"
+#include "time_machine.h"
+#include "bstests.h"
+
+#include <zephyr/types.h>
+#include <zephyr/sys/printk.h>
+
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/conn.h>
+
+extern enum bst_result_t bst_result;
+
+static struct bt_conn *g_conn;
+
+CREATE_FLAG(flag_ext_adv_seen);
+CREATE_FLAG(flag_connected);
+
+static void connected(struct bt_conn *conn, uint8_t err)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	if (err != BT_HCI_ERR_SUCCESS) {
+		FAIL("Failed to connect to %s: %u\n", addr, err);
+		bt_conn_unref(g_conn);
+		g_conn = NULL;
+		return;
+	}
+
+	printk("Connected to %s\n", addr);
+	SET_FLAG(flag_connected);
+}
+
+static void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	printk("Disconnected: %s (reason %u)\n", addr, reason);
+
+	bt_conn_unref(g_conn);
+	g_conn = NULL;
+}
+
+static void recycled(void)
+{
+	UNSET_FLAG(flag_connected);
```

Comment:
nitpick: I know this works, but for clarity could you add a `recycled` flag too just like for the advertiser?

You can even add a comment when waiting for it (in the test procedure below), to indicate that you have to wait for the only conn context to be deallocated for the scan start to be successful.

Reply:
Good catch. Added for consistency on both tests.